### PR TITLE
tools:Fix fpgaconf input file status

### PR DESCRIPTION
**Pull request by anandaravuri**
changes:
 - Check for input file is valid or not

### CODE REVIEW

File: tools/fpgaconf/fpgaconf.c [hunk 1]
```diff
@@ -389,10 +390,21 @@ int read_bitstream(char *filename, struct bitstream_info *info)
 	FILE *f;
 	long len;
 	int ret;
+	struct stat file_mode = { 0 };;
```
Comment (tswhison) [inline tools/fpgaconf/fpgaconf.c]: May want to memset file_mode to 0 instead of using the initializer syntax.  Klocwork is very picky about that.

Comment (luebbers) [reply tools/fpgaconf/fpgaconf.c]: And you only need one ';'. We have a limited supply. ;)

File: tools/fpgaconf/fpgaconf.c [hunk 2]
```diff
@@ -389,10 +390,21 @@ int read_bitstream(char *filename, struct bitstream_info *info)
...
 
 	if (!filename || !info)
 		return -EINVAL;
 
+	if (stat(filename, &file_mode) < 0) {
+		perror(filename);
+		return -1;
+	}
+
```
Comment (luebbers) [inline tools/fpgaconf/fpgaconf.c]: Klocwork may also call out that the attributes / existence of the file referred to by `filename` may change between the `stat()` call and a subsequent `open()` (similar, though not exactly identical to [this](https://wiki.sei.cmu.edu/confluence/display/c/POS35-C.+Avoid+race+conditions+while+checking+for+the+existence+of+a+symbolic+link)). You could possibly use `fstat()` instead after `open()`.

File: tools/fpgaconf/fpgaconf.c [hunk 3]
```diff
@@ -396,22 +396,22 @@ int read_bitstream(char *filename, struct bitstream_info *info)
...
+	if (!f) {
 		perror(filename);
 		return -1;
 	}
 
-	if (S_ISREG(file_mode.st_mode) == 0) {
-		fprintf(stderr, "Invalid input GBS file\n");
+	if (fstat(fileno(f), &file_mode) != 0) {
+		perror(filename);
 		return -1;
```
Comment (luebbers) [inline tools/fpgaconf/fpgaconf.c]: You need to `goto out_close` here.

File: tools/fpgaconf/fpgaconf.c [hunk 4]
```diff
@@ -396,22 +396,22 @@ int read_bitstream(char *filename, struct bitstream_info *info)
...
 
-	info->filename = filename;
-
-	/* open file */
-	f = fopen(filename, "rb");
-	if (!f) {
-		perror(filename);
+	if (S_ISREG(file_mode.st_mode) == 0) {
+		fprintf(stderr, "Invalid input GBS file\n");
 		return -1;
```
Comment (luebbers) [inline tools/fpgaconf/fpgaconf.c]: `goto out_close`

**Pull request merged**